### PR TITLE
DATAUP-449 add view configure mode for bulk import cells

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

Implements the "View Configure" tab that appears while jobs are running / completed.
Really, all that happens here is this creates a mode switch for the configure, file path, and params widgets. Default is "edit" mode, with "view" mode as an option when the flag is set. The only real difference is that "view" mode uses the `view` variant of the input widgets.

There's a couple of those widgets identified here that don't really play as expected. I suspect there are others, but these fixes fill the need at the moment. I'm sure there's be a bugfix jamboree at some point...

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-449
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* get a running bulk import cell, and click on "View Configure", this should show up with read only / disabled versions of all of the input widgets.
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js [hunk 1]
```diff
@@ -29,16 +29,16 @@ define([
      *  - workspaceId - the id of the workspace we should use for searching for objects
      *  - initialParams - Array of objects. Each item represents a row of parameter values.
      *    So each item has an object with parameter id -> value
+     *  - viewOnly - boolean, if true start with the view version of input widgets
      * @returns
      */
     function factory(config) {
+        const viewOnly = config.viewOnly || false;
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js:36]: Default's `false` if this doesn't show up in parameters.

File: kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js [hunk 2]
```diff
@@ -304,35 +287,40 @@ define([
          */
         function renderLayout() {
             let formContent = [];
+            const panelBody = [
+                ol({
+                    class: `${cssBaseClass}__list`,
+                    dataElement: `${cssClassType}-fields`,
+                }),
+            ];
+            if (!viewOnly) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js:296]: There should only be delete / add row buttons if we're not in viewOnly mode.

File: kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js [hunk 3]
```diff
@@ -430,8 +418,13 @@ define([
         function createFilePathWidget(rowId, appSpec, filePathParams, parameterId, parameterValue) {
             const spec = filePathParams.paramMap[parameterId];
             let widget;
-            return paramResolver
-                .loadInputControl(spec)
+            let controlPromise;
+            if (viewOnly) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js:422]: This repeated theme is the main change that makes viewOnly work.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js [hunk 4]
```diff
@@ -1,57 +1,42 @@
...
-    'bluebird',
-    'kb_common/html',
-    '../validation',
-    'common/events',
-    'common/runtime',
-    'common/dom',
-    'bootstrap',
-    'css!font-awesome',
-], (Promise, html, Validation, Events, Runtime, Dom) => {
+define(['bluebird', 'kb_common/html', 'common/dom', 'bootstrap', 'css!font-awesome'], (
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js:1]: Did some cleanup in this viewer. It was kinda... off.

File: nbextensions/bulkImportCell/bulkImportCell.js [hunk 5]
```diff
@@ -146,11 +146,11 @@ define([
                 tabs: {
                     configure: {
                         label: 'Configure',
-                        widget: ConfigureWidget,
+                        widget: ConfigureWidget.editMode,
```
Comment (briehl) [inline nbextensions/bulkImportCell/bulkImportCell.js:149]: The ConfigureWidget now implements 2 "widgets" - the edit and view versions. This lets the bulkImportCell.runTab not have to care about what widget its running and what mode it should be in.

File: kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js [hunk 6]
```diff
@@ -88,138 +84,141 @@ define([
                 referenceType: 'name',
                 paramsChannelName: bus.channelName,
                 closeParameters: closeParameters,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js:86]: another one that could be singularised (i.e. just use `closeParameters`)

it's not worth changing the PR just for this, though.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js [hunk 7]
```diff
@@ -1,57 +1,42 @@
...
+    Promise,
+    html,
+    Dom
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js:4]: Fairly sure that this `Dom` module replicates (some of) the contents of the UI module.

Then again, if you've got some cool code, you don't want to restrict it to just a single file -- copy it elsewhere!!

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js [hunk 8]
```diff
@@ -1,57 +1,42 @@
...
+            render();
         }
 
         function resetModelValue() {
             if (spec.data.defaultValue) {
-                setModelValue(spec.data.defaultValue);
+                return setModelValue(spec.data.defaultValue);
             } else {
-                unsetModelValue();
+                return unsetModelValue();
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js:39]: `else` is implied -- doesn't need to be written out

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js [hunk 9]
```diff
@@ -71,25 +56,18 @@ define([
...
 
         function render() {
-            Promise.try(() => {
-                const events = Events.make(),
-                    inputControl = makeInputControl(model.value, events, bus);
-
-                dom.setContent('input-container', inputControl);
-                events.attachEvents(container);
-            });
+            dom.setContent('input-container', makeInputControl(model.value));
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js:59]: Hmmm, suspiciously like `ui.setContent('data-element-name', content)`! :shipit:

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js [hunk 10]
```diff
@@ -1,57 +1,42 @@
...
-                    model.value = value;
-                    return true;
-                }
-                return false;
-            }).then((changed) => {
-                render();
-            });
+            if (model.value !== value) {
+                model.value = value;
+            }
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js:26]: may as well just do `model.value = value` and not bother with the equality check

File: nbextensions/bulkImportCell/bulkImportCell.js (hunk 5 above)
Comment (ialarmedalien) [reply nbextensions/bulkImportCell/bulkImportCell.js:149]: That's great! Be fantastic to port that over to the app cell and get rid of those vast chunks of copy-pasta

File: kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js (hunk 7 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js:4]: Yeah, I got a little sloppy with `newObjectView` and just did linting stuff. I'll clean it up a little further.

**Comment by briehl** [comment]
@ialarmedalien I rewrote that viewer widget to be a little more streamlined and use the UI module. Also added tests, as it didn't have a spec 😱

**Review by ialarmedalien — APPROVED** [review]
👍

**Pull request merged**